lib/node_modules/@stdlib/stats/nanmeanwd/README.md

```diff
@@ -0,0 +1,297 @@
+var zeros = require( '@stdlib/ndarray/zeros' );
+
+var x = array( [ 1.0, NaN, -2.0, 4.0 ] );
+var y = zeros( [] );
+
+var out = nanmeanwd.assign( x, y );
+// returns <ndarray>
+
+var v = out.get();
+// returns 1.0
+
+var bool = ( out === y );
+// returns true
+```
+
+The method has the following parameters:
+
+-   **x**: input [ndarray][@stdlib/ndarray/ctor]. Must have a real-valued or generic [data type][@stdlib/ndarray/dtypes].
+-   **out**: output [ndarray][@stdlib/ndarray/ctor].
+-   **options**: function options (_optional_).
+
+The method accepts the following options:
+
+-   **dims**: list of dimensions over which to perform a reduction. If not provided, the function performs a reduction over all elements in a provided input [ndarray][@stdlib/ndarray/ctor].
+
+</section>
+
+<!-- /.usage -->
+
+<section class="notes">
+
+## Notes
+
+-   This implementation uses Welford's algorithm for computing the arithmetic mean, providing improved numerical stability compared to naive summation approaches and being particularly beneficial when dealing with values of different magnitudes or accumulating floating-point errors.
+-   Setting the `keepdims` option to `true` can be useful when wanting to ensure that the output [ndarray][@stdlib/ndarray/ctor] is [broadcast-compatible][@stdlib/ndarray/base/broadcast-shapes] with ndarrays having the same shape as the input [ndarray][@stdlib/ndarray/ctor].
+-   The output data type [policy][@stdlib/ndarray/output-dtype-policies] only applies to the main function and specifies that, by default, the function must return an [ndarray][@stdlib/ndarray/ctor] having a real-valued floating-point or "generic" [data type][@stdlib/ndarray/dtypes]. For the `assign` method, the output [ndarray][@stdlib/ndarray/ctor] is allowed to have any supported output [data type][@stdlib/ndarray/dtypes].
+
+</section>
+
+<!-- /.notes -->
+
+<section class="examples">
+
+## Examples
+
+<!-- eslint no-undef: "error" -->
+
+```javascript
+var uniform = require( '@stdlib/random/base/uniform' );
+var filledarrayBy = require( '@stdlib/array/filled-by' );
+var bernoulli = require( '@stdlib/random/base/bernoulli' );
+var getDType = require( '@stdlib/ndarray/dtype' );
+var ndarray2array = require( '@stdlib/ndarray/to-array' );
+var ndarray = require( '@stdlib/ndarray/ctor' );
+var nanmeanwd = require( '@stdlib/stats/nanmeanwd' );
+
+function rand() {
+    if ( bernoulli( 0.8 ) < 1 ) {
+        return NaN;
+    }
+    return uniform( 0.0, 20.0 );
+}
+
+// Generate an array of random numbers:
+var xbuf = filledarrayBy( 25, 'generic', rand );
+
+// Wrap in an ndarray:
+var x = new ndarray( 'generic', xbuf, [ 5, 5 ], [ 5, 1 ], 0, 'row-major' );
+console.log( ndarray2array( x ) );
+
+// Perform a reduction:
+var y = nanmeanwd( x, {
+    'dims': [ 0 ]
+});
+
+// Resolve the output array data type:
+var dt = getDType( y );
+console.log( dt );
+
+// Print the results:
+console.log( ndarray2array( y ) );
+```
+
+</section>
+
+<!-- /.examples -->
+
```

Comment:
You should add the references for Welford's algorithm (see `stats/base/ndarray/nanmeanwd`).